app/views/universities/show.html.slim

```diff
@@ -0,0 +1 @@
+= page_content(@page).html_safe
```

Comment:
I'm currently plugging this into my site and realizing that I really dislike this way of rendering the pages.

I'm thinking that it makes more sense to do one of two options that are more Rails-y, and I'm interested in feedback on them:

## Layout template with direct page rendering

Have Frontmatter::PostsController:

1. Render the pages directly and assume that there's a template for pages instead of rendering into a show action:

   ```ruby
   def show
     # ...
     render "#{page_class.view_key}/#{@page.slug}", layout: "post"
   end
   ```

   ```erb
   <!-- a/v/layouts/post.html.erb -->
   <article class="post">
    <%= yield %>
   </article>
   ```

   ```markdown
   <!-- a/v/posts/my-post.md.yaml -->
   ---
   title: My Post
   date: 2020-01-01
   ---

   My first post
   ```
2. Provide the render bit in a controller-level helper method for overriding:

   ```ruby
   class PostsController < Frontmatter::PostsController
     def show
       # ...
       render_page
     end
   end
   ```

This method has the advantage that it uses familiar features such as layouts, with the drawback that it's surprising not to see a show view template for a show action.

## show view file with partial pages

Rather than using an application-level template (though it needn't be used elsewhere of course), use the show view file and render pages as partials. This necessitates that all pages be named `_<slug><extension>[other extensions]<.[yaml][json][toml]>` such as `_squash.md.erb.yaml`.

```erb
<!-- a/v/pages/show.html.erb -->
<article class="page show">
  <%= render @page %>
</article>
```

In this case, Frontmatter::Page would provide a to_partial_path of `<tableize>/<slug>` such as "universities/squash" for an object `#<University slug="squash">`

The advantage that the view files are all within a single directory and that the API is sensible without requiring new helpers is nice, I think. The drawbacks are that it requires renaming files with the leading underscore and that it overrides to_partial_path which could be surprising.

Do you have any thoughts on those as alternate APIs to the `page_content(@page).html_safe` helper? As it stands the API is a little closer to the "show view file with partial pages" option, but with additional API in the form of the helper.

Reply:
Because of template inheritance, Frontmatter can also provide a default, empty template or show page so that those pages don't need to exist if they're not wanted in an app.